### DISCUSSION TITLE
What is the coordinate system of the normal from AOV?

### DISCUSSION THREAD

**Discussion by GCChen97** [mitsuba-renderer/mitsuba3#309] [Q&A] [answered]
Hi,

Thanks for creating such a nice renderer. I want to use mitsuba for some research but I am confused about the definition of the coordinate system mitsuba uses. I have a few questions:

1. What is the definition of the camera frame in mitsuba? ( screen right==x positive, screen top==y positive? )
2. What is the coordinate system of the normal from AOV? In world frame or camera frame?
3. Is the coordinate system of mitsuba the same as blender? I used the blender addon `mitsuba-blender` to import scene.xml and noticed some differences of the two coordinate system.

**Comment by njroussel** [comment]
1. This might be confusing in writing, I'd recommend you check this on your own by using `scene.sensors()[0].world_transform().inverse()`. This gives you the transform from world space to the coordinate frame of the sensor.
The camera/sensor frame is set up as such: X=left, Y=up, Z=forward
2. The shading and geometry normals in the `aov` integrator are in the world frame.
3. I see you posted https://github.com/mitsuba-renderer/mitsuba-blender/discussions/47. I don't think there is a discrepancy between both, but I'll let them confirm that for you over there.

**Reply by GCChen97** [reply]
Thanks for the reply! I'll check this. Besides, I noticed that the ply meshes imported by the addon `mitsuba-blender` are -90 degree rotated around x axis comparing to those imported by `blender`. But obj meshes are consistently imported by both `mitsuba-blender` and `blender`. It is also confusing. Do you have any idea?

**Reply by njroussel** [reply]
I haven't used the Blender addon in quite some time. So, I can't tell you if this is intended behavior or not. Let's let @ros-dorian confirm all of this over at https://github.com/mitsuba-renderer/mitsuba-blender :smile:

**Comment by GCChen97** [comment, marked as answer]
Took some time figuring out the definitions:
1. In `mitsuba`, the camera frame is "screen left==x, screen top==y and forward==z", the world frame is "Y up and -Z forward".
2. In `blender`, the camera frame is "screen right==x, screen top==y and forward==-z", the world frame is "Z up and Y forward".
